package/boot/uboot-mediatek/patches/480-add-unielec-u7622-02.patch

```diff
@@ -0,0 +1,2188 @@
++			       "NDL6", "NDL7", "NRB";
++			input-enable;
++			bias-pull-up;
++		};
++
++		conf-clk {
++			pins = "NCLE";
++			bias-pull-down;
++		};
++
++	};
++
++	mmc1_pins_default: mmc1default {
++		mux {
++			function = "sd";
++			groups =  "sd_0";
++		};
++		/* "I2S2_OUT, "I2S4_IN"", "I2S3_IN", "I2S2_IN",
++		 *  "I2S4_OUT", "I2S3_OUT" are used as DAT0, DAT1,
++		 *  DAT2, DAT3, CMD, CLK for SD respectively.
++		 */
++		conf-cmd-data {
++			pins = "I2S2_OUT", "I2S4_IN", "I2S3_IN",
++			       "I2S2_IN","I2S4_OUT";
++			input-enable;
++			drive-strength = <8>;
++			bias-pull-up;
++		};
++		conf-clk {
++			pins = "I2S3_OUT";
++			drive-strength = <12>;
++			bias-pull-down;
++		};
++		conf-cd {
++			pins = "TXD3";
++			bias-pull-up;
++		};
++
++	};
++};
++
++&snor {
++        pinctrl-names = "default";
++        pinctrl-0 = <&snor_pins>;
++        status = "okay";
++
++        spi-flash@0{
++                compatible = "jedec,spi-nor";
++                reg = <0>;
++                spi-tx-bus-width = <1>;
++                spi-rx-bus-width = <4>;
++                u-boot,dm-pre-reloc;
++        };
++};
++
++&uart0 {
++	status = "okay";
++	mediatek,force-highspeed;
++};
++
++&pwm {
++	pinctrl-names = "default";
++	pinctrl-0 = <&pwm_pins>;
++	status = "okay";
++};
++
++&mmc0 {
++	pinctrl-names = "default";
++	pinctrl-0 = <&mmc0_pins_default>;
++	status = "okay";
++	bus-width = <8>;
++	max-frequency = <50000000>;
++	cap-mmc-highspeed;
++	vmmc-supply = <&reg_3p3v>;
++	vqmmc-supply = <&reg_3p3v>;
++	non-removable;
++};
++
++&mmc1 {
++	pinctrl-names = "default";
++	pinctrl-0 = <&mmc1_pins_default>;
++	status = "okay";
++	bus-width = <4>;
++	max-frequency = <12000000>;
++	cap-sd-highspeed;
++	r_smpl = <1>;
++	vmmc-supply = <&reg_3p3v>;
++	vqmmc-supply = <&reg_3p3v>;
++};
++
++&watchdog {
++	pinctrl-names = "default";
++	pinctrl-0 = <&watchdog_pins>;
++	status = "okay";
++};
++
++&eth {
++	status = "okay";
++	mediatek,gmac-id = <0>;
++	phy-mode = "sgmii";
++	mediatek,switch = "mt7531";
++	reset-gpios = <&gpio 54 GPIO_ACTIVE_HIGH>;
++
++	fixed-link {
++		speed = <1000>;
++		full-duplex;
++	};
++};
++
++&ssusb {
++	status = "okay";
++};
++
++&u3phy {
++	status = "okay";
++};
+--- a/arch/arm/mach-mediatek/mt7622/init.c
++++ b/arch/arm/mach-mediatek/mt7622/init.c
+@@ -5,11 +5,13 @@
+  */
+ 
+ #include <common.h>
+-#include <fdtdec.h>
+ #include <init.h>
++#include <asm/global_data.h>
+ #include <asm/armv8/mmu.h>
+ #include <asm/cache.h>
+ 
++DECLARE_GLOBAL_DATA_PTR;
++
+ int print_cpuinfo(void)
+ {
+ 	printf("CPU:   MediaTek MT7622\n");
+@@ -18,13 +20,10 @@
+ 
+ int dram_init(void)
+ {
+-	int ret;
+-
+-	ret = fdtdec_setup_memory_banksize();
+-	if (ret)
+-		return ret;
+-	return fdtdec_setup_mem_size_base();
++	gd->ram_size = get_ram_size((long *)CONFIG_SYS_SDRAM_BASE,
++					SZ_2G);
+ 
++        return 0;
+ }
+ 
+ void reset_cpu(void)
+--- a/configs/mt7622_unielec_u7622-02-emmc_defconfig
++++ b/configs/mt7622_unielec_u7622-02-emmc_defconfig
```

Comment:
The defconfig is not needed here. If you create defconfigs please use the `make savedefconfig ` from the kernel, it will minimize it.

Reply:
I don't see why the emmc_defconfig is not needed here.
I don't know the `make savedefconfig` can reduce the config size, Thanks. I will update the nor_defconfig.

Reply:
The defconfig is not used by OpenWrt, but we have to maintain this patch. I would prefer if you do not add these kernel defconfig files.